OneMore/Commands/File/Markdown/MarkdownConverter.cs

```diff
@@ -158,21 +195,43 @@ public MarkdownConverter RewriteTodo(IEnumerable<XElement> paragraphs)
 				{
 					var cdata = run.GetCData();
 					var wrapper = cdata.GetWrapper();
+					if (wrapper.FirstNode is XText)
+					{
+						cdata.Value = wrapper.GetInnerXml();
+					}
+					while (wrapper.FirstNode is not XText && wrapper.FirstNode is not null)
+					{
+						wrapper = (XElement)wrapper.FirstNode;
+					}
 					if (wrapper.FirstNode is XText text)
 					{
 						var match = boxpattern.Match(text.Value);
+						// special treatment of todo tag 
 						if (match.Success)
 						{
-							text.Value = text.Value.Substring(match.Length);
-
-							// ensure TagDef exists
-							var index = page.AddTagDef("3", "To Do", 4);
-
-							// inject tag prior to run
-							run.AddBeforeSelf(new Tag(index, match.Groups["x"].Value == "x"));
+							var org = text.Value;
+							var completed = match.Groups["x"].Value == "x";
+							text.Value = text.Value.Replace((completed ? "[x]" : "[ ]"), "");
+							cdata.Value = cdata.Value.Replace(org, text.Value);
+							page.SetTag(paragraph, tagSymbol: "3", tagStatus:completed,tagName:"todo");
+						}
+						else
+						{
+							// look for all other tags
+							foreach (var t in MarkdownEmojis.taglist)
+							{
+								// check for other tags
+								if (text.Value.Contains(t.name))
+								{
+									var org = text.Value;
+									text.Value = text.Value.Replace(t.name, "");
+									cdata.Value = cdata.Value.Replace(org, text.Value);
+									// ensure TagDef exists
+									page.SetTag(paragraph, tagSymbol: t.id, tagStatus: false, tagName: t.topic, tagType: t.type);
+									break;
```

Comment:
You're not looking for "all other tags", just the first one you find because of the break. Get ride of the foreach and just find the first using Linq